bpf/lib/lb.h

```diff
@@ -675,41 +675,44 @@ lb6_to_lb4_service(const struct lb6_service *svc __maybe_unused)
 }
 
 static __always_inline
-struct lb6_service *lb6_lookup_service(struct lb6_key *key,
-				       const bool scope_switch)
+struct lb6_service *__lb6_lookup_service(struct lb6_key *key)
 {
 	struct lb6_service *svc;
-	__u8 orig_proto = key->proto;
 
-	key->scope = LB_LOOKUP_SCOPE_EXT;
-	key->backend_slot = 0;
 	svc = map_lookup_elem(&cilium_lb6_services_v2, key);
 
 #if defined(ENABLE_SERVICE_PROTOCOL_DIFFERENTIATION)
 	/* If there are no elements for a specific protocol, check for ANY entries. */
-	if (!svc && key->proto != 0) {
-		key->proto = 0;
+	if (!svc && key->proto != IPPROTO_ANY) {
+		__u8 orig_proto = key->proto;
+
+		key->proto = IPPROTO_ANY;
 		svc = map_lookup_elem(&cilium_lb6_services_v2, key);
+		key->proto = orig_proto;
```

Comment:
There's a *very* subtle regression here - we're flipping back to the `orig_proto` even on success, which means that the caller can't just re-use this `lb*_key` struct to look up the backend slot in the service map (because the `proto` between frontend and backend entry won't match).

And I'm very suspicious what changed over the last few weeks in the datapath, so that the socket LB test now actually caught this. And why it didn't cause trouble before.